src/rules/maxLineLengthRule.ts

```diff
@@ -44,20 +69,52 @@ export class Rule extends Lint.Rules.AbstractRule {
         return `Exceeds maximum line length of ${lineLimit}`;
     }
 
-    public isEnabled(): boolean {
-        return super.isEnabled() && this.ruleArguments[0] as number > 0;
-    }
-
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as number);
+        return this.applyWithFunction(sourceFile, walk, parseOptions(this.ruleArguments));
     }
 }
 
-function walk(ctx: Lint.WalkContext<number>) {
-    const limit = ctx.options;
+function walk(ctx: Lint.WalkContext<Options>) {
     for (const line of getLineRanges(ctx.sourceFile)) {
-        if (line.contentLength > limit) {
-            ctx.addFailureAt(line.pos, line.contentLength, Rule.FAILURE_STRING_FACTORY(limit));
+        if (
+            line.contentLength > ctx.options.limit &&
+            !(
+                ctx.options.ignoreUrls &&
+                hasUrl(line, ctx.sourceFile) &&
+                lineLengthMinusUrl(line, ctx.sourceFile) < ctx.options.limit
+            )
+        ) {
+            ctx.addFailureAt(
+                line.pos,
+                line.contentLength,
+                Rule.FAILURE_STRING_FACTORY(ctx.options.limit),
+            );
         }
     }
 }
+
+function hasUrl(line: LineRange, sourceFile: ts.SourceFile): boolean {
+    return urlRegex().test(sourceFile
+                            .text.substr(line.pos, line.contentLength));
+}
+
+function isOptions(arg: number | Options): arg is Options {
+    return (arg as Options).limit !== undefined;
+}
+
+function lineLengthMinusUrl(line: LineRange, sourceFile: ts.SourceFile): number {
+    return sourceFile
+            .text
+            .substr(line.pos, line.contentLength)
+            .replace(urlRegex(), "")
+            .length;
```

Comment:
I'm wondering what users would expect us to do here.
Consider a limit of 140 and a line with length 180. The line includes 2 urls with 20 characters each.
IMO the rule should still complain. There should be an exception for **long** urls that cannot be broken apart. But short urls can be moved to a separate line without a problem.
I'm thinking about a threshold of 1/2 or 1/3 of the limit.
What do you think?

Reply:
I agree the rule should still complain, however it may be confusing for users: they would explicitly config the rule to `{ ignoreUrls: true }`, but the rule may still complain about URLs sometimes. This could be fixed with something like renaming the option to `ignoreLongUrls`?

Reply:
You're right. Your suggestion makes sense.
Does it make sense to allow users to configure what is considered "long"? For example `{ignore(Long)Urls: 100}`
On the other hand we would repeat that configuration when adding an option to ignore long regular expression literals.

Reply:
I'm on the fence. Part of me thinks this is within the scope of a "sensible" default setting. That is, we pick a setting--the `100` from your last example--and enforce it quietly?

Reply:
Agreed. Let's just got with a sensible default and see how it pans out.
We shouldn't use a fixed limit though. I know there are people using a limit of 80 for max line length. Let's just use 1/2 of `limit` to also allow some indentation on the same line.